core/shared/src/main/scala/zio/FiberRef.scala

```diff
@@ -29,10 +29,25 @@ package zio
  *
  * `result` will be equal to "Hi!" as changes done by child were merged on join.
  *
+ * FiberRef also forms a monoid using the combine function provided in FiberRef#make.
```

Comment:
Does the operation need to form a `Monoid` or just a `Semigroup`? It seems like we need an associative operation `combine` but don't actually need a `zero` value since the `FiberRef` will always be filled with an initial value on creation and children will inherit that. For instance, in all your examples you use `0` as the value but that isn't actually a `zero` value for `max` over the integers. I don't think it matters for the code but just how we describe it.

Reply:
I'd say it forms (or should form) a monoid with the initial value as the 'zero'. Obviously we can't enforce that and to avoid scaring people away, the monoid thing should be a footnote rather than a headliner.

Also the examples we use should probably use the "real" zero for the provided combine

Reply:
What if, say, we want to create a `FiberRef` with an initial value of one and addition as the operation to count the number of fibers that an effect creates?

Reply:
Also our default implementation of `combine` is `last` which doesn't have a `zero`.

Reply:
Good point. Maybe just leave this unspecified. It's more general than a monoid, but well-defined really only for monoids.

Reply:
You are right. I was thinking of the initial value as the zero value for monoid when writing this, but this does not work as it does not always serve as the identity. I'll update

Reply:
Oh, I updated on a stale version of the comments, this was directly replying to @adamgfraser initial comment. I'll take this out of the docs